cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/update/StateVariablesAdder.java

```diff
@@ -100,20 +203,99 @@ public static void add(Network n, CgmesModel cgmes) {
                 tapSteps.add(p);
             }
         }
-        cgmes.add(CgmesSubset.STATE_VARIABLES, "SvTapStep", tapSteps);
+
+        for (ThreeWindingsTransformer t : network.getThreeWindingsTransformers()) {
+            PropertyBag p = new PropertyBag(svTapStepProperties);
+            Arrays.asList(t.getLeg1(), t.getLeg2(), t.getLeg3()).forEach(leg -> {
+                if (leg.getPhaseTapChanger() != null) {
+                    p.put(tapChangerPositionName, is(leg.getPhaseTapChanger().getTapPosition()));
+                    p.put(CgmesNames.TAP_CHANGER, cgmes.phaseTapChangerForPowerTransformer(t.getId()));
+                    tapSteps.add(p);
+                } else if (leg.getRatioTapChanger() != null) {
+                    p.put(tapChangerPositionName, is(leg.getRatioTapChanger().getTapPosition()));
+                    p.put(CgmesNames.TAP_CHANGER, cgmes.ratioTapChangerForPowerTransformer(t.getId()));
+                    tapSteps.add(p);
+                }
+            });
+        }
+
+        cgmes.add(originalSVcontext, "SvTapStep", tapSteps);
+    }
+
+    private void addStatusToCgmes() {
+        // create SvStatus, iterate on Connectables, check Terminal status, add
+        // to SvStatus
+        PropertyBags svStatus = new PropertyBags();
+        network.getVoltageLevelStream()
+            .flatMap(VoltageLevel::getConnectableStream)
+            .distinct()
+            .forEach(c -> {
+                PropertyBag p = new PropertyBag(SV_SVSTATUS_PROPERTIES);
+                for (Terminal t : ((Connectable<?>) c).getTerminals()) {
+                    p.put(IN_SERVICE, Boolean.toString(t.isConnected()));
+                }
+                p.put(CgmesNames.CONDUCTING_EQUIPMENT, c.getId());
+                svStatus.add(p);
+            });
+
+        // SvStatus at boundaries set as it was in original cgmes.
+        for (PropertyBag terminal : originalTerminals) {
+            Objects.requireNonNull(terminal);
```

Comment:
Are you sure this check is needed: in which case the terminal is null?

Reply:
Removed unnecessary check